protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -526,8 +559,9 @@ func schemaOfField(f *descriptor.Field, reg *descriptor.Registry, refs refMap) o
 		if wktSchema, ok := wktSchemas[fd.GetTypeName()]; ok {
 			core = wktSchema
 
-			if fd.GetTypeName() == ".google.protobuf.Empty" {
-				props = &openapiSchemaObjectProperties{}
+			// if wfkSchema is a reference to another known type, add it to refs to be processed later
+			if core.Ref != "" {
+				refs[fd.GetTypeName()] = struct{}{}
```

Comment:
I'm not sure I understand this change, could you explain?

Reply:
I can't remember exactly, however I believe this was the previous behaviour:

`rpcStatus` (which references `protobufAny`) is hardcoded to always output in the `definitions` object, and `rpcStatus.details` is an array of `"$ref": "#/definitions/protobufAny"`.

If you define a type in your API with a `google.protobuf.Any` field, the generated schema outputted the schema as an explicit schema under that properties object, and not as a `$ref`. This meant if your schema referenced `Any` multiple times, your schema would define the schema for `Any` in every usage, growing your schema.

---

Because our `wktSchemas` maps now includes `.google.protobuf.Any` as a known-type with the ref field populated, any references to `Any` in a custom type will be declared using said `$ref`. However we need to make sure that the type is still outputted to `definitions`, so we add it to the `refs` map so that it is added during the call to `addCustomRefs`.